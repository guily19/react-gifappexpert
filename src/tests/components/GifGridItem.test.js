import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GifGridItem from '../../components/GifGridItem';

describe('<GifGridItem />', () => {

  it('renders with title', () => {
    const props = {
      title: 'image title',
      url: 'testurl.com'
    } 
    const wrapper = shallow(<GifGridItem {...props}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('title must be the p element', () => {
    const props = {
      title: 'image title',
      url: 'testurl.com'
    } 
    const wrapper = shallow(<GifGridItem {...props}/>);
    const pElement = wrapper.find('p');
    expect(pElement.text()).toEqual(props.title);
  });

  it('image has to be url and title from props', () => {
    const props = {
      title: 'image title',
      url: 'testurl.com'
    } 
    const wrapper = shallow(<GifGridItem {...props}/>);
    const imgElement = wrapper.find('img');
    expect(imgElement.prop('alt')).toEqual(props.title);
    expect(imgElement.prop('src')).toEqual(props.url);
  });

  it('div has to be at least class card', () => {
    const props = {
      title: 'image title',
      url: 'testurl.com'
    } 
    const wrapper = shallow(<GifGridItem {...props}/>);
    const divElement = wrapper.find('div');
    const className = divElement.prop('className');
    expect(className.includes('card')).toBe(true);
  });


});