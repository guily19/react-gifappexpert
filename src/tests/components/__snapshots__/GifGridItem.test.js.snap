// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> renders with title 1`] = `
<div
  className="card"
>
  <img
    alt="image title"
    src="testurl.com"
  />
  <p>
    image title
  </p>
</div>
`;
